v2/lints/etsi/lint_qcstatem_qclimitvalue_valid.go

```diff
@@ -53,38 +45,36 @@ func isOnlyLetters(s string) bool {
 }
 
 func (l *qcStatemQcLimitValueValid) Execute(c *x509.Certificate) *lint.LintResult {
+	errString := util.ErrorStringBuilder{Delimiter: "; "}
+
+	if len(c.QCStatements.ParsedStatements.Limit) != 1 {
+		return &lint.LintResult{Status: lint.Error, Details: "invalid number of MonetaryValue objects"}
+	}
+	qcLv := c.QCStatements.ParsedStatements.Limit[0]
 
-	errString := ""
-	ext := util.GetExtFromCert(c, util.QcStateOid)
-	s := util.ParseQcStatem(ext.Value, *l.getStatementOid())
-	errString += s.GetErrorInfo()
-	if len(errString) == 0 {
-		qcLv, ok := s.(util.EtsiQcLimitValue)
-		if !ok {
-			return &lint.LintResult{Status: lint.Error, Details: "parsed QcStatem is not a EtsiQcLimitValue"}
+	if qcLv.Amount < 0 {
+		errString.Append("amount is negative")
+	}
+
+	// Check whether the alphabetic currency code was set
+	if len(qcLv.Currency) > 0 {
```

Comment:
I was talking with @dadrian this morning about this. I think we need to/should change ZCrypto first. As mentioned on #434 , we need to disambiguate which CHOICE was selected. The easiest way to do this is to update ZCrypto such that `Currency` is a `*string` and `CurrencyAmount` is a `*int`, emphasizing effectively that either are optional, so that we can nil-test to distinguish between an 'empty' currency and a valid one.

I was also discussing whether it would be appropriate to have ZCrypto more closely align with the ASN.1 grammar here so that it can do things like check string type (`PrintableString`) and constraint length.

But I think we should resolve those ZCrypto issues, which is why I started poking at #434 - to suss them out.